test/unit/test_mechcat.cpp

```diff
@@ -244,11 +245,24 @@ TEST(mechcat, fingerprint) {
         arb::fingerprint_mismatch);
 }
 
+TEST(mechcat, loading) {
+    EXPECT_THROW(load_catalogue("does/not/exist-catalogue.so"), dynamic_catalogue_error);
+    try {
+        auto cat = load_catalogue(LIBDIR "/dummy-catalogue.so");
+        EXPECT_EQ(std::vector<std::string>{"dummy"}, cat.mechanism_names());
+    }
+    catch (arb::dynamic_catalogue_error& e) {
+        std::cerr << e.what() << '\n';
+        e.print_platform_error(std::cerr);
+        throw;
+    }
+}
```

Comment:
For checking error states, this could be expanded with `EXPECT_THROW` and `ASSERT_NO_THROW`, but again, this sort of coverage is spotty in the whole unit test suite.